src/colmap/geometry/homography_matrix.cc

```diff
@@ -202,9 +247,17 @@ void PoseFromHomographyMatrix(const Eigen::Matrix3d& H,
 
   points3D->clear();
   std::vector<Eigen::Vector3d> tentative_points3D;
+  double best_reproj_error_sum = std::numeric_limits<double>::max();
   for (size_t i = 0; i < cams2_from_cams1.size(); ++i) {
-    CheckCheirality(cams2_from_cams1[i], points1, points2, &tentative_points3D);
-    if (tentative_points3D.size() >= points3D->size()) {
+    // Note that we can typically eliminate 2 of the 4 solutions using the
```

Comment:
This was the main underlying issue. Before we would randomly pick 1 of the 2 remaining solutions but it was wrong half of the times.